### ISSUE TITLE
Problems and feedback on the Avalonia.MusicStore tutorial

### ISSUE THREAD

**Issue by ruilvo** [AvaloniaUI/Documentation#57]
Hello there,

I recently followed the Avalonia.MusicStore tutorial and there are some quirks that bothered me throughout the experience. 

I decided to compile everything I noticed, and this will be a dump of that. I hope they can be addressed.

Thank you for both the Avalonia product and these tutorials!

Now, onto business:

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/add-content-to-dialog

In 
> Add a BuyMusicCommand property like so:

The first code block is okay, but then in:

> In the contructor add the following:

This constructor hasn't been defined as of yet.

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/add-content-to-dialog

> Lets test this by adding a contructor on the MusicStoreViewModel.cs.

I had to add the constructor before for the `BuyMusicCommand = ReactiveCommand.CreateFromTask(...` stuff...
So it's not `adding a constructor` so much as `adding *to* the constructor`.

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/return-from-dialog

> The Button of the MusicStoreView has its Command property bound to BuyMusicCommand. 
> This doesn't exist yet so lets add this to MusicStoreViewModel with the following code.

But we already added this in a previous step. Seems that these are in the wrong order.

> Now add a constructor to MusicStoreViewModel where we can instantiate the command, 
> and implement the code needed to return a result from the dialog.

The snippet contains:

```xaml
BuyMusicCommand = ReactiveCommand.Create(() =>
{
	return SelectedAlbum;
});
```

But previously we were already presented with a much more complete version:

```xaml
BuyMusicCommand = ReactiveCommand.CreateFromTask(async () =>
{
	if (SelectedAlbum is { })
	{
		return SelectedAlbum;
	}
	return null;
});
```

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/return-from-dialog

In 

> We pass to that a Subscription to our ViewModels BuyMusicCommand. We subscribe it directly to the Windows Close method.

Might be worth remembering to add `using System;`. Intellisense was the one reminding me.

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/displaying-images

In

> Firstly open AlbumViewModel.cs and add a property of type Bitmap? 
> named Cover using the usually pattern so that it can notify the UI when it changes.

It would help to also remember to add `using Avalonia.Media.Imaging;`. 
In this case, Intellisense didn't know if I wanted `Avalonia.Media.Imaging` or some
random `System.*` thing. 

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/add-items-to-users-collection

In

> Modify the MainWindow.axaml so that it places the existing Button inside a panel. 

But we did this since the start...

In the following code snippet, it doesn't remind the reader to add

```xaml
        xmlns:views="clr-namespace:Avalonia.MusicStore.Views"
        xmlns:local="clr-namespace:Avalonia.MusicStore.Views"
```

In the `Window` properties.

-----------------------------------------------------------------------------------------

https://docs.avaloniaui.net/tutorials/music-store-app/load-data-at-startup

In

> Add a method to MainWindowViewModel.cs like:

Again, it's missing a `using Avalonia.MusicStore.Models;`.

Then the tutorial ends here. The `LoadAlbums` isn't ever used, so there is no loading of albums at startup...

**Comment by daniel-lerch** [comment]
I have fixed your first three suggestions in #62 and will work on the remaining points in the next days.

**Comment by daniel-lerch** [comment]
From your last point I left out reminding the user to add `using Avalonia.MusicStore.Models;` because this one is trivial and there are many other using directives required which are also not mentioned.

The other using directives you mentioned are rather tricky. In my case Visual Studio's IntelliSense wasn't even able to suggest them correctly. That's why I addressed these points in #65.
